source/agora/network/Manager.d

```diff
@@ -853,6 +854,11 @@ public class NetworkManager
 
     public void getUnknownTXs (NodeLedger ledger) @safe nothrow
     {
+        // Protect against running multiple times simultaneously
+        if (this.isGettingUnknownTXs)
```

Comment:
What happened if a race condition occurs? The `getUnknownTXs` is called in two places, so I think it might happen.

Reply:
Good point, I'll add `synchronized`.

Reply:
Updated with `shared bool` and `synchronized` block for checking and setting the value.

Reply:
Reverted as they are called from the same thread.